test/graphql_test.rb

```diff
@@ -374,4 +374,35 @@ def test_mutation_batch_subselection_execution
     assert_equal expected, result
     assert_equal ["Product/1,2", "Product/2,3"], queries
   end
+
+  class MyCustomExecutor < GraphQL::Batch::Executor
+    attr_reader :loaders_executed
+
+    def initialize
+      @loaders_executed = []
+      super
+    end
+
+    def around_promise_callbacks(loader)
+      @loaders_executed << loader
+      super
+    end
+  end
+
+  def test_custom_executor_class
```

Comment:
I think this should move to another test class, since this one gets re-used in GraphQL::ExecutionStrategyTest which tests the deprecated code that we don't support custom executors for.